.github/workflows/buildpack/test-image.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+set -e
+
+if [ -z "${IMAGE_NAME}" ] || [ -z "${IMAGE_VARIANT}" ]; then
+  echo "\${IMAGE_NAME} and \${IMAGE_VARIANT} need to be defined."
+  exit 1
+fi
+
+SCRIPT=/tmp/scripts/build.sh
+if [ -f ".github/workflows/buildpack/test/${IMAGE_VARIANT}.sh" ]; then
+  SCRIPT="/tmp/.github/workflows/buildpack/test/${IMAGE_VARIANT}.sh"
+fi
+
+if [ "${IMAGE_VARIANT}" == "emscripten" ]; then
+  # build_emscripten.sh will setup docker and executes build within
+  scripts/build_emscripten.sh emscripten_build "${IMAGE_NAME}"
```

Comment:
``scripts/build_emscripten.sh`` just runs another script in a docker image... so if you just unify the fact that the solidity repo is mounted to ``/root/project`` in everywhere and create test scripts for each image (symlinking them, if they'd be the same), this would reduce to (together with the other suggestions)

docker run -v "${REPO_ROOT}:/root/project" current-image /root/project/.github/workflows/buildpack/test-${IMAGE_KIND}.sh

with the ``else`` case below being the same.